### PR TITLE
feat(Napier): Add top-level function and some changes

### PR DESCRIPTION
**Pull request by ghasemdev**
**Changes:** 
- [x] Add Kdoc for Napier and update README.md
- [x] Add a new function for wrapping Napier object.
- [x] Change the input order of the log functions for better readability, especially for the lambda at the end.
- [x] Fix duplicated issue.

Please accept pull request.

### CODE REVIEW

**Comment by ghasemdev** [comment]
> @ghasemdev I really appreciate your help. This is nice!
> 
> Please fix the test and swift code.
> 
> * ios/Napier/Napier.swift
> 
> Change the `message_`.
> 
> ```swift
> shared.log(
>             priority: logLevel,
>             tag: tag ?? {
>                 let fileName = URL(fileURLWithPath: file).lastPathComponent
>                 let functionName: String
>                 if let firstBraceIndex = function.firstIndex(of: "(") {
>                     functionName = String(function[..<firstBraceIndex])
>                 } else {
>                     functionName = function
>                 }
>                 return "\(fileName):\(functionName)"
>             }(),
>             throwable: nil,
>             message: message
>         )
> ```

Tanks, Also I fixed that problem.

File: napier/src/commonMain/kotlin/io/github/aakira/napier/Napier.kt [hunk 1]
```diff
@@ -26,87 +63,190 @@ object Napier {
...
     }
 
+    /** Send a ERROR log message and log the exception.
+     *
+     * @param tag String: Used to identify the source of a log message. It usually
+     * identifies the class or activity where the log call occurs. This value may be null.
+     * @param throwable Throwable: An exception to log This value may be null.
+     * @param message String: The message you would like logged. This value cannot be null.
+     */
     fun e(message: String, throwable: Throwable? = null, tag: String? = null) {
```
Comment (AAkira) [inline napier/src/commonMain/kotlin/io/github/aakira/napier/Napier.kt:161]: Here is the reason why the test does not work.

```kotlin
fun e(message: String, tag: String? = null, throwable: Throwable? = null) 
```

Comment (AAkira) [reply napier/src/commonMain/kotlin/io/github/aakira/napier/Napier.kt:161]: @ghasemdev Please fix this.

**Comment by AAkira** [comment]
Sorry for pointing this out again and again.
Does it need to reorder the arguments of the previously defined function?

The sample project doesn't work now.

* `mpp-sample/src/commonMain/.../Sample.kt`

```kotlin
    fun handleError() {
        try {
            throw Exception("throw error")
        } catch (e: Exception) {
            Napier.e("Napier Error", e) // error
        }
    }
```

The order of the arguments should remain the same as before because I think there are only a few situations where a tag is used.

**Comment by ghasemdev** [comment]
> Sorry for pointing this out again and again. Does it need to reorder the arguments of the previously defined function?
> 
> The sample project doesn't work now.
> 
> * `mpp-sample/src/commonMain/.../Sample.kt`
> 
> ```kotlin
>     fun handleError() {
>         try {
>             throw Exception("throw error")
>         } catch (e: Exception) {
>             Napier.e("Napier Error", e) // error
>         }
>     }
> ```
> 
> The order of the arguments should remain the same as before because I think there are only a few situations where a tag is used.

I think you are right. I did not notice that we can configure the tag through the DebugAntilog class. I'm sorry that most of the tests lead to errors. I don't have Android SDK in the IntelliJ and the project dependencies are not installed properly and I can't  run the tests.

I fix issue and commit again.

**Comment by ghasemdev** [comment]
I have another question, This enum class is un useless?

image

**Comment by AAkira** [comment]
Thank you so mush!
LGTM

> I have another question, This enum class is un useless?

I forgot to delete it in this [PR](https://github.com/AAkira/Napier/pull/53).
I'll fix it in another PR.
Thank you.

**Pull request merged**